lib/omniauth/strategies/shopify.rb

```diff
@@ -23,6 +23,31 @@ def valid_site?
         !!(/\A(https|http)\:\/\/[a-zA-Z0-9][a-zA-Z0-9\-]*\.#{Regexp.quote(options[:myshopify_domain])}[\/]?\z/ =~ options[:client_options][:site])
       end
 
+      def valid_signature?
+        return false unless request.POST.empty?
+
+        params = request.GET
+        signature = params['hmac']
+        timestamp = params['timestamp']
+        return false unless signature && timestamp
+
+        return false unless timestamp.to_i > Time.now.to_i - 5 * 60
```

Comment:
Oops, I meant to write this in the PR instead of the commit.

Do we want to disallow future timestamps as well? If a server has a very slow clock, then nothing will fail but it will be vulnerable to replays.